src/Language/PureScript/Errors.hs

```diff
@@ -1318,7 +1319,7 @@ toTypelevelString t = (Box.text . decodeStringWithReplacement) <$> toTypelevelSt
   toTypelevelString' :: Type -> Maybe PSString
   toTypelevelString' (TypeLevelString s) = Just s
   toTypelevelString' (TypeApp (TypeConstructor f) x)
-    | f == primName "TypeString" = Just $ fromString $ prettyPrintType x
+    | f == primName "TypeString" = fromString <$> (initMay . Box.render $ typeAsBox x)
```

Comment:
All the pretty printing operations rely on the box model. Are you suggesting coming up with an alternative pretty printer for types?